### PR TITLE
Workspace bar changes

### PR DESCRIPTION
**Pull request by matavach**
Two changes:

1.  Add a masking feature to workspace numbers to normalize them across multiple monitors.
   - For example, I have 20 total workspaces across 2 monitors. With this, both my monitors display 1-10 for workspace numbers.
   - Only works with "workspace numbers" and  "monitor specific" enabled
   - Recommended to be used with something like [this](https://github.com/jasper-at-windswept/hypr-ws-switcherl) for allowing seamless keybinds across monitors
   
2. Added option to display neither a highlight or underline for the active workspace.

### CODE REVIEW

File: widget/settings/pages/config/bar/index.ts [hunk 1]
```diff
@@ -33,12 +33,13 @@ export const BarSettings = () => {
                 Option({ opt: options.bar.workspaces.icons.active, title: 'Workspace Active', type: 'string' }),
                 Option({ opt: options.bar.workspaces.icons.occupied, title: 'Workspace Occupied', type: 'string' }),
                 Option({ opt: options.bar.workspaces.show_numbered, title: 'Show Workspace Numbers', type: 'boolean' }),
-                Option({ opt: options.bar.workspaces.numbered_active_indicator, title: 'Numbered Workspace Identifier', subtitle: 'Only applicable if Workspace Numbers are enabled', type: 'enum', enums: ["underline", "highlight"] }),
+                Option({ opt: options.bar.workspaces.numbered_active_indicator, title: 'Numbered Workspace Identifier', subtitle: 'Only applicable if Workspace Numbers are enabled', type: 'enum', enums: ["underline", "highlight", "color only"] }),
```
Comment (Jas-SinghFSU) [inline widget/settings/pages/config/bar/index.ts:36]:
```suggestion
                Option({ opt: options.bar.workspaces.numbered_active_indicator, title: 'Numbered Workspace Identifier', subtitle: 'Only applicable if Workspace Numbers are enabled', type: 'enum', enums: ["underline", "highlight", "color"] }),
```
As this is an enumerated value, only is implied by nature so we can leave it out.

File: widget/settings/pages/config/bar/index.ts [hunk 2]
```diff
@@ -33,12 +33,13 @@ export const BarSettings = () => {
...
                 Option({ opt: options.theme.bar.buttons.workspaces.numbered_active_highlight_border, title: 'Highlight Radius', subtitle: 'Only applicable if Workspace Numbers are enabled', type: 'string' }),
                 Option({ opt: options.theme.bar.buttons.workspaces.numbered_active_text_color, title: 'Highlighted Text Color', subtitle: 'Only applicable if Workspace Numbers are enabled', type: 'color' }),
                 Option({ opt: options.bar.workspaces.spacing, title: 'Spacing', subtitle: 'Spacing between workspace icons', type: 'float' }),
                 Option({ opt: options.bar.workspaces.workspaces, title: 'Total Workspaces', type: 'number' }),
                 Option({ opt: options.bar.workspaces.monitorSpecific, title: 'Monitor Specific', subtitle: 'Only workspaces applicable to the monitor will be displayed', type: 'boolean' }),
+                Option({ opt: options.bar.workspaces.workspaceMask, title: 'Mask Workspace Numbers On Monitors', subtitle: 'Only applicable if Workspace Numbers and Monitor Specific are enabled.\nMasks workspaces per monitor to keep workspace numbers consistent across all monitors.', type: 'boolean' }),
```
Comment (Jas-SinghFSU) [inline widget/settings/pages/config/bar/index.ts:42]:
```suggestion
                Option({ opt: options.bar.workspaces.workspaceMask, title: 'Mask Workspace Numbers On Monitors', subtitle: 'Only applicable if Workspace Numbers and Monitor Specific are enabled.\nForces each Monitor's Workspaces to start counting from 1.' }),
```

Comment (Jas-SinghFSU) [reply widget/settings/pages/config/bar/index.ts:42]: Let's shorten the title and make the subtitle more descriptive to the behavior of the setting.

**Comment by Jas-SinghFSU** [comment]
And since the subtitle/description for the option is 2 lines, it stretches the switch vertically which makes it look weird. We can add `vpack: "center"` to remedy this for the `boolean` case in the `Inputter.ts` file as show here:
```js
                case "boolean": return self.child = Widget.Switch({
                    vpack: "center", // lets add this
                    sensitive: disabledBinding !== undefined ? disabledBinding.bind("value").as(disabled => !disabled) : true,
                })
```

File: modules/bar/workspaces/index.ts [hunk 3]
```diff
@@ -148,8 +155,8 @@ const Workspaces = (monitor = -1, ws = 8) => {
         component: Widget.Box({
             class_name: "workspaces",
             children: Utils.merge(
-                [workspaces.bind(), monitorSpecific.bind()],
-                (workspaces, monitorSpecific) => {
+                [workspaces.bind(), monitorSpecific.bind(), workspaceMask.bind()],
+                (workspaces, monitorSpecific, workspaceMask) => {
```
Comment (Jas-SinghFSU) [inline modules/bar/workspaces/index.ts:159]:
```suggestion
                [workspaces.bind(), monitorSpecific.bind()],
                (workspaces, monitorSpecific) => {
```

File: modules/bar/workspaces/index.ts [hunk 4]
```diff
@@ -215,7 +222,9 @@ const Workspaces = (monitor = -1, ws = 8) => {
                                                     return available;
                                                 }
                                             }
-                                            return `${i}`;
```
Comment (Jas-SinghFSU) [inline modules/bar/workspaces/index.ts:218]: It won't let me suggest a code change but let's add the binds to this Utils.merge scope.
```js
                                    label: Utils.merge(
                                        [
                                            options.bar.workspaces.show_icons.bind("value"),
                                            options.bar.workspaces.icons.available.bind("value"),
                                            options.bar.workspaces.icons.active.bind("value"),
                                            options.bar.workspaces.icons.occupied.bind("value"),
                                            workspaceMask.bind("value"),
                                            hyprland.active.workspace.bind("id"),
                                        ],
                                        (showIcons, available, active, occupied, workspaceMask, _) => {
                                            if (showIcons) {
                                                if (hyprland.active.workspace.id === i) {
                                                    return active;
                                                }
                                                if ((hyprland.getWorkspace(i)?.windows || 0) > 0) {
                                                    return occupied;
                                                }
                                                if (
                                                    monitor !== -1
                                                ) {
                                                    return available;
                                                }
                                            }
                                            return workspaceMask
                                                ? `${index + 1}`
                                                : `${i}`;
                                        },
                                    ),

```

**Comment by Jas-SinghFSU** [comment]
Looks good otherwise :+1:

**Comment by matavach** [comment]
Did some testing on adding `vpack: "center"` option in different spots on the Inputter class.

Vpack center on both Inputter box and child switch box (both switch and border are centered)
image
Child box only (hard to tell but the border fills the container while the switch is centered):
image
Inputter Box Only (same as both parent and child):
image
Neither (border fills its container, and switch fills border):
image

From this, I'm going to add `vpack: "center"` to the Inputter parent box only, since it seems that adding it to both the parent and child is redundant. This also normalizes centering for all inputter instances.

I went through the rest of the config to confirm this didn't break anything, and I wasn't able to find any issues with making this change global for the class.

File: modules/bar/workspaces/index.ts [hunk 5]
```diff
@@ -199,9 +206,10 @@ const Workspaces = (monitor = -1, ws = 8) => {
                                             options.bar.workspaces.icons.available.bind("value"),
                                             options.bar.workspaces.icons.active.bind("value"),
                                             options.bar.workspaces.icons.occupied.bind("value"),
+                                            workspaceMask.bind(),
```
Comment (Jas-SinghFSU) [inline modules/bar/workspaces/index.ts:209]:
```suggestion
                                            workspaceMask.bind("value"),
```
Let's bind the value of the variable here.

**Pull request merged**